src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -0,0 +1,365 @@
+                var fieldNumber = staticValidationAttributesDict.Count + 1;
+                staticValidationAttributeInstance = new StaticFieldInfo(
+                    FieldTypeFQN: attr.AttributeName,
+                    FieldOrder: fieldNumber,
+                    FieldName: $"A{fieldNumber}",
+                    InstantiationLines: attrInstantiationStatementLines);
+
+                staticValidationAttributesDict.Add(instantiationStatement, staticValidationAttributeInstance);
+            }
+
+            return staticValidationAttributeInstance;
+        }
+
+        private void GenTransitiveValidation(ValidatedMember vm, ref Dictionary<string, StaticFieldInfo> staticValidatorsDict)
+        {
+            string callSequence;
+            if (vm.TransValidateTypeIsSynthetic)
+            {
+                callSequence = vm.TransValidatorType!;
+            }
+            else
+            {
+                var staticValidatorInstance = GetOrAddStaticValidator(ref staticValidatorsDict, vm.TransValidatorType!);
+
+                callSequence = $"{StaticValidatorHolderClassFQN}.{staticValidatorInstance.FieldName}";
+            }
+
+            var valueAccess = (vm.IsNullable && vm.IsValueType) ? ".Value" : string.Empty;
+
+            if (vm.IsNullable)
+            {
+                OutLn($"if (options.{vm.Name} != null)");
+                OutLn($"{{");
+                OutLn($"    builder.AddResult({callSequence}.Validate(baseName + \"{vm.Name}\", options.{vm.Name}{valueAccess}));");
+                OutLn($"}}");
+            }
+            else
+            {
+                OutLn($"builder.AddResult({callSequence}.Validate(baseName + \"{vm.Name}\", options.{vm.Name}{valueAccess}));");
+            }
+        }
+
+        private void GenEnumerationValidation(ValidatedMember vm, ref Dictionary<string, StaticFieldInfo> staticValidatorsDict)
+        {
+            var valueAccess = (vm.IsValueType && vm.IsNullable) ? ".Value" : string.Empty;
+            var enumeratedValueAccess = (vm.EnumeratedIsNullable && vm.EnumeratedIsValueType) ? ".Value" : string.Empty;
+            string callSequence;
+            if (vm.EnumerationValidatorTypeIsSynthetic)
+            {
+                callSequence = vm.EnumerationValidatorType!;
+            }
+            else
+            {
+                var staticValidatorInstance = GetOrAddStaticValidator(ref staticValidatorsDict, vm.EnumerationValidatorType!);
+
+                callSequence = $"{StaticValidatorHolderClassFQN}.{staticValidatorInstance.FieldName}";
+            }
+
+            if (vm.IsNullable)
+            {
+                OutLn($"if (options.{vm.Name} != null)");
+            }
+
+            OutOpenBrace();
+
+            OutLn($"var count = 0;");
+            OutLn($"foreach (var o in options.{vm.Name}{valueAccess})");
+            OutOpenBrace();
+
+            if (vm.EnumeratedIsNullable)
+            {
+                OutLn($"if (o is not null)");
```

Comment:
For object members, `!= null` is used, but for enumerables, `is not null` is used. Is that intentional?

Reply:
I tend to prefer `is null` and `is not null` in places where `o` can be any type just in case the type includes weirdly defined equality operators.